build.gradle

```diff
@@ -165,7 +166,7 @@ subprojects {
             mavenDeployer {
                 beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
 
-                repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
+                repository(url: publishingRepositoryUrl) {
                     authentication(userName: publishingUsername, password: publishingPassword)
```

Comment:
> One question is there a way to avoid to set ossrhPassword, ossrhUsername

To make `publishingUsername` and `publishingPassword` optional.
Try something like
```groovy
if (publishingUsername != '' && publishingPassword != '') {
  authentication(userName: publishingUsername, password: publishingPassword)
}
```


> and so no ossrhUrl to a local file instead into the gradle.properties ?

http://mrhaki.blogspot.com/2010/09/gradle-goodness-different-ways-to-set.html

Reply:
> Environment variables or setting the value in ~/.gradle/gradle.properties are two options that work well by default with gradle.

Oh yes sorry I didn't tried this one !

After there an other need you didn't defined the snapshot repository, this could help ! 
see example at the end of the doc http://central.sonatype.org/pages/gradle.html
 if you want i can provide the link but i won't be able to test it !